packages/core/src/components/TextField/TextField.tsx

```diff
@@ -176,11 +179,12 @@ const TextField: VibeComponent<TextFieldProps, unknown> & {
     );
 
     const onChangeCallback = useCallback(
-      (value: string) => {
+      (value: string, e?: React.ChangeEvent<HTMLInputElement>) => {
         if (isRequiredAndEmpty && value) {
           setIsRequiredAndEmpty(false);
         }
-        onChange(value, { target: inputRef.current });
+        const event = e || { target: inputRef.current };
```

Comment:
Can e be ever undefined?

Reply:
It would be undefined when the "clear" button is clicked on the Search variant (if you're asking me, special cases like this, should be same as I did in the new Search - in the Search component and not in the Input, which should be as generic as possible.)